chore(docs): updated outdated auth docs (#2534)
botpress/botpress: docs/guide/docs/advanced/authentication.md

title: Authentication Methods
---

There's three different types of authentication:

1. **Collaborators**: These users are able to access the Admin UI and will have access to manage and edit bots based on their roles.
2. **Chat Users**: These users are only able to speak with bots, and they can see the list of bots if they log on the Admin UI
3. **External User**: This method of authentication expects that you handle user authentication yourself and provide a JWT token to identify the user.

## Authentication Overview

> There are 4 different type of authentication strategies currently supported: Basic, SAML, OAUTH2 and LDAP. You can also implement them multiple times (for example, you could have two different OAuth2 configurations for different workspaces).

Reviewer: ```suggestion
There are 4 different type of authentication strategies currently supported: Basic, SAML, OAuth2 and LDAP. You can also implement them multiple times (for example, you could have two different OAuth2 configurations for different workspaces).
```

No matter what authentication strategy is used, they are all stored in the database. When you add a new strategy in the `botpress.config.json` file, upon server restart a new table called `strategy_STRATEGYID` will be created.

When you give access to a user for a specific workspace, an entry is created in the table `workspace_users` with his role.

If you have more than one authentication strategy, a menu will be displayed to pick a strategy. You can skip the menu and bookmark a specific strategy by changing the page URL: `/admin/login/STRATEGYID`.

Moreover, you can access a specific workspace by using `/admin/login?workspaceId=WORKSPACEID`

> https://github.com/botpress/botpress/blob/master/src/bp/core/config/botpress.config.ts#L326

Reviewer: ```suggestion
You can find the definition for the various authentication strategies [here](https://github.com/botpress/botpress/blob/master/src/bp/core/config/botpress.config.ts#L326).
```

### Basic

Basic Authentication allows a user to log in with a simple username / password. The password is salted for added security.

To create more accounts, visit the `Collaborators` tab on the Admin UI. Choose the role and enter the E-mail of your collaborator, then you will receive a random password. The user will need to pick a password after the first login.

Super Admins are able to reset the password of any user using the basic authentication.

> Here's a basic configuration example to set it up:

Reviewer: ```suggestion
#### Configuration Example
In your `botpress.config.json` file:
```

```js
{
 "pro": {
    "collaboratorsAuthStrategies": ["default"],
  },
  "authStrategies": {
    "default": {
      "type": "basic",
      "label": "Default Auth",
      "options": {}
    }
  }
}
```

#### Additional Security

> There are additional options that can be configured when using that authentication strategy. Please refer to the [configuration file for more informations](https://github.com/botpress/botpress/blob/master/src/bp/core/config/botpress.config.ts#L350) :

Reviewer: ```suggestion
There are additional options that can be configured when using this authentication strategy. Please refer to the [configuration file for more information](https://github.com/botpress/botpress/blob/master/src/bp/core/config/botpress.config.ts#L350) :
```

> - maxLoginAttempt: Max number of tries allowed before locking out the user

Reviewer: ```suggestion
- `maxLoginAttempt`: Max number of tries allowed before locking out the user
```

> - lockoutDuration: Account will be disabled for this amount of time when maxLoginAttempt is reached

Reviewer: ```suggestion
- `lockoutDuration`: Account will be disabled for this amount of time when `maxLoginAttempt` is reached
```

Reviewer: Also, what values can this propertie take? seconds? milliseconds? It's not clear when only reading the docs.

> - passwordExpiryDelay: Password will expire after this specified duration

Reviewer: ```suggestion
- `passwordExpiryDelay`: Password will expire after this specified duration
```

> - passwordMinLength: Minimum length for the user's password

Reviewer: ```suggestion
- `passwordMinLength`: Minimum length for the user's password
```

> - requireComplexPassword: Requires at least 1 characters of 3 category of characters

Reviewer: ```suggestion
- `requireComplexPassword`: Requires at least 1 characters of 3 category of characters
```
